src/protocol-handlers/urls.js

```diff
@@ -0,0 +1,147 @@
+const toUri = require('multiaddr-to-uri')
+const { CID } = require('multiformats/cid')
+const fetch = require('node-fetch')
+
+const DEFAULT_GATEWAY = 'https://dweb.link'
+const LOCAL_HOSTNAMES = ['127.0.0.1', '[::1]', '0.0.0.0', '[::]']
+
+/**
+ * @typedef ParsedUrl
+ * @type {object}
+ * @property {string} protocol
+ * @property {string} hostname
+ * @property {string} path
+ */
+
+/**
+ * Parses an IPFS/IPNS/dWeb URL to be handled by IPFS Desktop.
+ *
+ * @param {string} url
+ * @returns {ParsedUrl}
+ */
```

Comment:
I see a `return null` which means `parseUrl` does not only return a `ParsedUrl` type. Be sure to include null in the possible returned values.

I believe a simple `// @ts-check` pragma at the top of the page would display this problem if your editor is configured properly.